hw/dv/sv/sim_sram/index.md

```diff
@@ -0,0 +1,80 @@
+## Simulation SRAM
+
+The `sim_sram` module intercepts an outbound TL interface to carve out a chunk of "fake" memory used for simulation purposes only.
+This chunk of memory must not overlap with any device on the system address map - it must be an invalid address range from the system's perspective.
+
+It has 4 interfaces - clock, reset, input TL interface and the output TL interface.
+It instantiates a [1:2 TL socket]({{< relref "hw/ip/tlul/doc#bus-primitives" >}}) to split the incoming TL access into two.
+One of the outputs of the socket is sinked within the module while the other is returned back.
+
+If the user chooses to instantiate an actual SRAM instance, the sinked TL interface is connected to TL SRAM adapter which converts the TL access into an SRAM access.
+The [technology-independent]({{< relref "hw/ip/prim/README" >}}) `prim_ram_1p` module is used for instantiating the memory.
+
+![Block Diagram](sim_sram.svg)
+
+This module is not meant to be synthesized.
+Though, it is written with the synthesizable subset of SystemVerilog to be Verilator-friendly.
+An easy way to ensure this is to instantiate it ONLY if either `VERILATOR` or `UVM` macro is defined.
+
+The most typical usecase for this module is for a SW test running on an embedded CPU in the design to be able to exchange information with the testbench.
+In DV, this is envisioned to be used for the following purposes:
+- write the status of the SW test
+- use as a SW logging device
+- write the output of an operation
+- signal an event
+
+These usecases apply to Verilator as well.
+However, at this time, the Verilator based simulations rely on the on-device UART for logging.
+
+This module must be instantiated before any XBAR / bus fabric element in the design which can return an error response when it sees an access to an invalid address range.
+Ideally, it should be placed as close to the CPU (or equivalent) as possible to reduce the simuation time.
+
+## Customizations
+
+`sim_sram` exposes the following parameters:
+
+- `InstantiateSram` (default: 0):
+
+  Controls whether to instantiate the SRAM or not.
+  The most typical (and a very much recommended) operating mode is for the SW test to only write data to this SRAM, to make it portable across various simulation platforms (DV, Verilator, FPGA etc.).
+  The testbench can simply probe the sinked output of the socket to monitor writes to specific addresses within this range.
+  So, in reality the actual SRAM instance is not even needed.
+  However, there may be some future usecases for instantiating it.
```

Comment:
Do we actually need this in the patch? If we can't think of any reason to do it at the moment, I'd suggest only adding the functionality if we need it in the future. ("Don't generalise too early, because you'll always generalise in the wrong direction" and all that)